File: packages/go/templates.go
package hubfly

import "context"

type TemplatesService struct {
	client *Client
}

// List returns all available application templates
func (s *TemplatesService) List(ctx context.Context) (*Response[[]Template], error) {
	return Request[[]Template](ctx, s.client, "GET", "/api/v1/templates", nil)
}

File: packages/ts/src/modules/templates.ts
import type { HubflyClient } from '../client.js';
import type { ApiResponse, Template } from '../types.js';

export class TemplatesModule {
  constructor(private readonly client: HubflyClient) {}

  /**
   * List available application templates
   */
  public async list(): Promise<ApiResponse<Template[]>> {
    return this.client.request<Template[]>('/api/v1/templates');
  }
}
